src/compiler/checker.ts

```diff
@@ -13335,6 +13335,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
 
+            if (isInJSFile(declaration)) {
+                const thisTag = getJSDocThisTag(declaration);
+                if (thisTag && thisTag.typeExpression) {
+                    thisParameter = createSymbolWithType(createSymbol(SymbolFlags.FunctionScopedVariable, InternalSymbolName.This), getTypeFromTypeNode(thisTag.typeExpression));

```

Comment:
Would it be possible to add `bindJSDocThisTag` in the binder instead of creating a symbol here in the checker? 
I'm not sure what you'd use as the declaration -- I guess the tag name, or maaaybe the tag itself.

I'm not 100% sure it would work, but if it doesn't, this in an OK fallback.

Reply:
I was thinking about creating a symbol in a `binder`, and I struggled with several questions.

1. What do we need to use as a `parent`? Do we need it?
2. What do we need to use as a `declaration`?
    > I guess the tag name, or maaaybe the tag itself.

    I think we can only use `tagName`, the tag itself is not a _Declaration_. Or create a special kind of `Declaration` that will contain a `typeExpression` and `tagName`.
3. How to determine the type of this `symbol`?
    The `Declaration` of `tagName` doesn't refer to `typeExpression`. Therefore in the `checker`, we need to change `getTypeOfSymbol` to resolve the type from `typeExpression`. Something like so,
   
```ts
// binder
function bindJSDocThisTag() {
 // todo
}

// getSignatureFromDeclaration
if (isInJSFile(declaration)) {
    const thisTag = getJSDocThisTag(declaration);
    if (thisTag && thisTag.tagName.symbol) {
        thisParameter = thisTag.tagName.symbol;
    }
}

// getTypeOfVariableOrParameterOrPropertyWorker
const typeExpression = /*find typeExpression*/
if (symbol.escapedName === InternalSymbolName.This && typeExpression) {
     return getTypeFromTypeNode(typeExpression) 
}
```

Reply:
@sandersn Could you look at my previous comment?

Reply:
@sandersn Could you look at my previous comment?

Reply:
From poking around the binder, I convinced myself that the symbol must be created in the checker.
It can't be created in the binder, because a parameter's symbol needs to have a parameter as a declaration, and there is no `this` parameter in a JS file.

The closest tag that does create a declaration is `@template {Constraint} T`, but `T` is not a type, it's a TypeParameterDeclaration, which is usable as a declaration. `@this` has only a type, as you point out.

Sorry for the delay on this one.